eqcorrscan/core/match_filter.py

```diff
@@ -605,12 +608,15 @@ def write(self, filename, format='tar'):
                               format='QUAKEML')
             for i, family in enumerate(self.families):
                 print('Writing family %i' % i)
-                _write_family(
-                    family=family, filename=filename + os.sep +
-                    family.template.name + '_detections.csv')
+                name = family.template.name + '_detections.csv'
+                name_to_write = join(filename, name)
+                _write_family(family=family, filename=name_to_write)
             with tarfile.open(filename + '.tgz', "w:gz") as tar:
                 tar.add(filename, arcname=os.path.basename(filename))
-            shutil.rmtree(filename)
+            if os.path.isdir(filename):
```

Comment:
A test incorporating this if statement would be good - does look like it needs to be here - thanks!

Reply:
I think I just had to put this in to make the tests past; somehow I got in a state where they were failing because it was trying to delete a directory that didn't exist or was a file instead. Looking closer at the code, it does seem strange this can happen given line 600 should make the directory. I am not really sure what is going on here but I will try to look into it more.

Reply:
Agreed, there should be a directory - this is one instance that using the `tempfile.TemporaryDirectory` context manager would be really handy, but isn't available in py2.7 AFAIK. 
It really shouldn't ever be a file eh... Weird, I don't know how this could have happened, but if you need help let me know.